web/toolbar.js

```diff
@@ -0,0 +1,282 @@
+
+/**
+ * @typedef {Object} ToolbarOptions
+ * @property {HTMLDivElement} container - Container for the secondary toolbar.
+ * @property {HTMLSpanElement} numPages - Label that contains number of pages.
+ * @property {HTMLInputElement} pageNumber - Control for display and user input
+ *   of the current page number.
+ * @property {HTMLSpanElement} scaleSelectContainer - Container where scale
+ *   controls are placed. The width is adjusted on UI initialization.
+ * @property {HTMLSelectElement} scaleSelect - Scale selection control.
+ * @property {HTMLOptionElement} customScaleOption - The item to display of the
+ *   non-predefined scale.
+ * @property {HTMLButtonElement} previous - Button to go to the previous page.
+ * @property {HTMLButtonElement} next - Button to go to the next page.
+ * @property {HTMLButtonElement} zoomIn - Button to zoom in the current page.
+ * @property {HTMLButtonElement} zoomOut - Button to zoom out the current page.
+ * @property {HTMLButtonElement} viewFind - Button to open find bar.
+ * @property {HTMLButtonElement} openFile - Button to for opening file dialog.
+ * @property {HTMLButtonElement} presentationModeButton - Button to switch to
+ *   the presentation mode.
+ * @property {HTMLButtonElement} download - Button to download the document.
+ * @property {HTMLAElement} viewBookmark - Element to link current url of
+ *   the page view.
+ */
+
+/**
+ * @class
+ */
+var Toolbar = (function ToolbarClosure() {
+  /**
+   * @constructs Toolbar
+   * @param {ToolbarOptions} options
+   * @param {HTMLDivElement} mainContainer
+   * @param {EventBus} eventBus
+   */
+  function Toolbar(options, mainContainer, eventBus) {
+    this.toolbar = options.container;
+    this.mainContainer = mainContainer;
+    this.eventBus = eventBus;
+    this.items = options;
+
+    this._wasLocalized = false;
+    this.reset();
+
+    // Bind the event listeners for click and hand tool actions.
+    this._bindListeners();
+  }
+
+  Toolbar.prototype = {
+    setPageNumber: function (pageNumber, pageLabel) {
+      this.pageNumber = pageNumber;
+      this.pageLabel = pageLabel;
+      this._updateUIState(false);
+    },
+
+    setPagesCount: function (pagesCount, hasPageLabels) {
+      this.pagesCount = pagesCount;
+      this.hasPageLabels = hasPageLabels;
+      this._updateUIState(true);
+    },
+
+    setPageScale: function (pageScaleValue, pageScale) {
+      this.pageScaleValue = pageScaleValue;
+      this.pageScale = pageScale;
+      this._updateUIState(false);
+    },
+
+    reset: function () {
+      this.pageNumber = 0;
+      this.pageLabel = null;
+      this.hasPageLabels = false;
+      this.pagesCount = 0;
+      this.pageScaleValue = DEFAULT_SCALE_VALUE;
+      this.pageScale = DEFAULT_SCALE;
+      this._updateUIState(true);
+    },
+
+    _bindListeners: function Toolbar_bindClickListeners() {
+      var eventBus = this.eventBus;
+
+      this.items.previous.addEventListener('click', function() {
```

Comment:
Nit: perhaps a tiny bit less verbose to define `var items = this.items` above, and use that here (and it the other cases below) instead. I don't feel to strongly about it though!